lib/cache/cache.js

```diff
@@ -25,13 +25,12 @@ const assert = require('assert')
  * @typedef {[any, any][]} requestResponseList
  */
 
-class Cache {
-  /**
-   * The name of the cache
-   * @type {string}
-   */
-  #id
+/**
+ * @type {(o: Cache) => requestResponseList}
+ */
+let getCacheRequestResponseList
```

Comment:
This module global seems very very dangerous and a source of leaks.

Reply:
node has a similar pattern in its mimetype parser, https://github.com/nodejs/node/blob/d3b0a2a68b5cb03de1f254cefa46f4c4023b1f89/lib/internal/mime.js#L296

Reply:
I could change to a symbol if you want, this is just more private (Object.getOwnPropertySymbols returns an empty array)

Reply:
Node.js usage is not a cache. This looks like an unbounded list.

Reply:
The list generally has to be unbounded - every time .open is called a new Cache is created with the requestResponseList associated with the cache name. Basically:

```js
// the two caches are different instances
const c1 = await caches.open('v1')
const c2 = await caches.open('v1')

assert.notEqual(c1, c2) // same name, different instances
assert.deepStrictEqual(await c1.keys(), await c2.keys()) // same cache names
assert.deepStrictEqual(await c1.match('/req'), await c2.match('/req'))
```

Reply:
I guess we can store the Cache instances themselves and then access the list inside the constructor...

Reply:
So you did remove the global cache, am I correct?

Reply:
where in the code is this logic? I can't find it.

I don't think we can support this feature in Node.js, as this would result in a massive memory leak.

Reply:
https://github.com/nodejs/undici/pull/2076/files#diff-138705835e0968882bd15c371854df97c448d16e5ef1352ede0f57634df917caR57-R74

In a future iteration we'd use a file based cache in place of this, right now it's just trying to match the spec.

Reply:
That code does not have a global list of caches or response, right?

Reply:
no it doesn't